sequence_processing_pipeline/FastQCJob.py

```diff
@@ -0,0 +1,135 @@
+        # SBATCH -p SOMETHING -> PBS -q SOMETHING
+        # (Torque doesn't appear to have a quality of service (-q) option. so
+        # this will go unused in translation.)
+        lines.append("#PBS -q %s" % queue_name)
+
+        # request one node
+        # Slurm --ntasks-per-node=<count> -> -l ppn=<count>	in Torque
+        lines.append("#PBS -l nodes=%d:ppn=%d" % (node_count, nprocs))
+
+        # Slurm --export=ALL -> Torque's -V
+        lines.append("#PBS -V")
+
+        # Slurm
+        # walltime limit --time=24:00:00 -> Torque's -l walltime=<hh:mm:ss>
+        # using the larger value found in the two scripts (36 vs 24 hours)
+        lines.append("#PBS -l walltime=%d:00:00" % wall_time_limit)
+
+        # send an email to the list of users defined below when a job starts,
+        # terminates, or aborts. This is used to confirm that the package's
+        # own reporting mechanism is reporting correctly.
+        lines.append("#PBS -m bea")
+
+        # list of users to be contacted independently of this package's
+        # notification system, when a job starts, terminates, or gets aborted.
+        lines.append("#PBS -M "
+                     "ccowart@ucsd.edu,"
+                     "jdereus@ucsd.edu,"
+                     "qiita.help@gmail.com")
+
+        # min mem per CPU: --mem-per-cpu=<memory> -> -l pmem=<limit>
+        # taking the larger of both values (10G > 6G)
+        # lines.append("#PBS -l pmem=10gb")
+
+        # --output -> -o
+        lines.append("#PBS -o %s" % self.stdout_log_path.format(x, y))
+        lines.append("#PBS -e %s" % self.stderr_log_path.format(x, y))
+
+        # there is no equivalent for this in Torque, I believe
+        # --cpus-per-task 1
+
+        # By default, PBS scripts execute in your home directory, not the
+        # directory from which they were submitted. Use run_dir instead.
+        lines.append("set -x")
+        lines.append("date '+%s' > /{}/{}.log".format(self.run_dir, job_name))
+        lines.append("source ~/miniconda3/bin/activate test_env_2")
+        lines.append("module load fastp_0.20.1 samtools_1.12 minimap2_2.18")
+        lines.append("cd %s" % self.run_dir)
+        # lines.append("export PATH=$PATH:/usr/local/bin")
+        lines.append("#file=${trim_file}\\${SLURM_ARRAY_TASK_ID}")
+        lines.append('%s %s %s %s %s %s' % (
+            self.fastqc_path, self.run_dir, self.output_dir, self.nprocs,
+            self.project, chemistry))
+        lines.append("echo $? >> /%s/%s.log" % (self.run_dir, job_name))
+        lines.append("date '+%s' >> /{}/{}.log".format(
+            self.run_dir, job_name))
+
+        with open(self.job_script_path, 'w') as f:
+            logging.debug("Writing job script to %s" % self.job_script_path)
+            for line in lines:
+                # remove long spaces in some lines.
+                line = re.sub(r'\s+', ' ', line)
+                f.write("%s\n" % line)
+
+    def run(self):
+        pass
```

Comment:
Does this override a superclass method? If so, is it correct for run() to be a noop?